Calculator/Calculator/Model/ExpressionParser.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  ExpressionParser.swift
+//  Calculator
+//
+//  Created by Daehoon Lee on 2023/06/06.
+//
+
+enum ExpressionParser {
+    static func parse(from input: String) -> Formula {
+        var formula: Formula = Formula(operands: CalculatorItemQueue<Double>(),
+                                       operators: CalculatorItemQueue<Operator>())
+        let operands = componentsByOperators(from: input).compactMap { Double($0) }
+        let operatorCandidates: String = input.hasPrefix("-") ? String(input.suffix(input.count - 1)) : input
+        let operators = operatorCandidates
+            .replacingOccurrences(of: "+-", with: "+")
+            .replacingOccurrences(of: "--", with: "-")
+            .replacingOccurrences(of: "/-", with: "/")
+            .replacingOccurrences(of: "*-", with: "*")
+            .compactMap { Operator(rawValue: $0) }
```

Comment:
```suggestion
        let operators = Operator.allCases.reduce(operatorCandidates) {
            return $0.replacingOccurrences(of: "\($1.rawValue)-", with: "\($1.rawValue)")
        }.compactMap { Operator(rawValue: $0) }
```

`componentsByOperators`에서 사용한 것과 같이 이것도 `Operator,allCases`를 활용해볼 수 있을 것 같습니다.

Reply:
안내해 주신 방법으로 수정하였습니다.👍